src/views/CamoView.vue

```diff
@@ -0,0 +1,123 @@
+			<a href="https://github.com/carlssonemil/orion/issues/new">GitHub</a>. Thanks and good luck
+			with the grind! ✌
+		</AlertComponent>
+		<FiltersComponent />
+		<CamosComponent :camos="filteredCamos" />
+		<ProgressComponent />
+	</div>
+</template>
+
+<script>
+import { mapState } from 'pinia'
+import { useStore } from '@/stores/store'
+import { groupBy, filterObject } from '@/utils/utils'
+import allCamos from '../data/camos'
+
+import AlertComponent from '@/components/AlertComponent.vue'
+import FiltersComponent from '@/components/FiltersComponent.vue'
+import CamosComponent from '@/components/CamosComponent.vue'
+import ProgressComponent from '@/components/ProgressComponent.vue'
+
+export default {
+	components: {
+		AlertComponent,
+		FiltersComponent,
+		CamosComponent,
+		ProgressComponent,
+	},
+
+	computed: {
+		...mapState(useStore, ['weapons', 'filters']),
+
+		filteredCamos() {
+			let filteredWeapons = this.weapons
+			const { hideGold, hidePlatinum, hidePolyatomic, hideNonRequired, category } = this.filters
+
+			if (hideNonRequired) {
+				filteredWeapons = filteredWeapons.filter((weapon) => !weapon.dlc)
+			}
+
+			if (hideGold) {
+				filteredWeapons = filteredWeapons.filter(
+					(weapon) =>
+						!Object.values(filterObject(weapon.progress, ['Platinum', 'Polyatomic'])).every(Boolean)
+				)
+			}
+
+			if (hidePlatinum) {
+				filteredWeapons = filteredWeapons.filter(
+					(weapon) => !Object.values(filterObject(weapon.progress, ['Polyatomic'])).every(Boolean)
+				)
+			}
+
+			if (hidePolyatomic) {
+				filteredWeapons = filteredWeapons.filter(
+					(weapon) => !Object.values(weapon.progress).every(Boolean)
+				)
+			}
+
+			if (category && category !== 'null') {
+				filteredWeapons = filteredWeapons.filter((weapon) => weapon.category === category)
+			}
+			let allCamoProgress = filteredWeapons.map((item) => item.progress)
+			//Can filter by % mastery here
+
+			// Flattens the object array
+			allCamoProgress = allCamoProgress.reduce(function (acc, x) {
+			for (var key in x) acc[key] = x[key]
+				return acc
+			}, {})
+
+			var allCamoCategories = [
+				'Spray Paint',
+				'Woodland',
+				'Digital',
+				'Dragon',
+				'Geometric',
+				'Fun',
+				'Foliage',
+				'Skulls',
+				'Tiger',
+				'Stripes',
+				'Reptile',
+				'Solid Colors',
+				'Classic',
+				'Cliffside',
+			]
+			var filteredCamos = allCamos.filter(
+				(camo) =>
+					allCamoCategories.some((s) => s == camo.category) &&
+					Object.keys(allCamoProgress).some((s) => s == camo.name)
+			)
+			filteredCamos = filteredCamos.map((camo) => {
+				camo.isCompleted = allCamoProgress[camo.name] ?? false
+				return camo
+			})
+			
+			var returnVal = groupBy(filteredCamos, (camo) => camo.category)
+			return returnVal
+		},
+	},
+}
+</script>
+
+<style lang="scss" scoped>
+.container {
+	text-align: center;
+}
+
+h1 {
+	margin-top: 75px;
+}
+
+h2 {
+	margin: 30px auto 0;
+	max-width: 450px;
+}
+</style>
```

Comment:
Let's remove the filtering for now. I'll make the `FilterComponent` more dynamic so it fits better 😄 The alert is probably not needed here either.
```suggestion
<template>
	<div class="container">
		<CamouflagesComponent :camouflages="camouflages" />
		<ProgressComponent />
	</div>
</template>

<script>
import { mapState } from 'pinia'
import { useStore } from '@/stores/store'
import { groupBy, filterObject } from '@/utils/utils'
import allCamos from '../data/camos'

import CamouflagesComponent from '@/components/CamouflagesComponent.vue'
import ProgressComponent from '@/components/ProgressComponent.vue'

export default {
	components: {
		CamosComponent,
		ProgressComponent,
	},

	computed: {
		...mapState(useStore, ['weapons']),
		
		camouflages() {
		  return groupBy(allCamos, (camouflage) => camouflage.category)
		},
	},
}
</script>

<style lang="scss" scoped>
.container {
	text-align: center;
}

h1 {
	margin-top: 75px;
}

h2 {
	margin: 30px auto 0;
	max-width: 450px;
}
</style>
```

Reply:
I've cleaned up the filtering and just left it with the initialisation code for getting the camouflage entities and progress. Yeah we can probably refactor the filtering code to be shared, I wasn't sure on best practices for that in vue and how they handle shared methods when working with active state.